_posts/2020-07-28-newpipe-0.19.8-released.md

```diff
@@ -0,0 +1,42 @@
+---
+layout: post
+title: "NewPipe 0.19.7 and 0.19.8 released: YouTube decrypt fix and many improvements"
+short: "NewPipe 0.19.8 released"
+date: 2020-07-27 22:00:00 +0200
+categories: [release, pinned]
+author: tobigr
+image: newpipe
+excerpt_separator: <!-- more -->
+---
+
+YouTube is back working again! Apart from that, NewPipe 0.19.7 and 0.19.8 come with auto-play for all services and numerous fixes.
+
+<!-- more -->
+
+## Once again: YouTube fixes
+
+Ah - what a night! Did you sleep well? Well, we didn't. On Monday, YouTube decided to change some things in their player again. These kinds of changes typically modify how the data of contained streams is extracted. As a result, NewPipe and other services like youtube-dl or invidious cannot access the stream and video data. Way too many of you have seen the result: nothing works.
+
+The folks at youtube-dl were quick and [delivered a fix incredibly fast](https://github.com/ytdl-org/youtube-dl/commit/e450f6cb634f17fd4ef59291eafb68b05c141e43). [@TobiGr](https://github.com/TobiGr) converted the fix from Python to Java and used [@wb9688](https://github.com/wb9688)'s work from a previous pull request to improve the handling of the patterns to decrypt the player info.
+We were able to release a new version of NewPipe Extractor, the library which gathers all the info you can see in the app, at 2am. Unfortunately, this did not mean, we could release a new version of NewPipe straight away. Programmers sign their applications so devices and users can verify they have an official version which has not been compromised by a third party. In our case, signing is done by [@TheAssassin](https://github.com/TheAssassin). However, he was not available in the morning and luckily managed to sign the created APK later in the afternoon. That created the big delay between creating a fix and releasing a new version. We hope, you can understand that. We did not want to let you wait an extra time with a broken NewPipe version ;)
+
+Apart from that, [@wb9688](https://github.com/wb9688) fixed second page search results and related videos not loading as well as age restricted videos not showing any info. Yes - YouTube was busy with changing stuff lately and so were we. 
```

Comment:
This is the third or fourth time I have to tell you that nothing regarding related videos has changed. The issue with age restricted videos is that they don't have that info anymore, and not having info causes `Exception`s and thus a crash report. I also added different methods of extracting certain things, but return simply `""` or `-1` for others.